app/addons/config/components.react.jsx

```diff
@@ -0,0 +1,410 @@
+//  Licensed under the Apache License, Version 2.0 (the "License"); you may not
+//  use this file except in compliance with the License. You may obtain a copy of
+//  the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+//  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+//  License for the specific language governing permissions and limitations under
+//  the License.
+
+import React from "react";
+import ReactDOM from "react-dom";
+import Stores from "./stores";
+import Actions from "./actions";
+import {Overlay, Button, Popover} from "react-bootstrap";
+import Components from "../components/react-components.react";
+import FauxtonComponents from "../fauxton/components.react";
+
+var configStore = Stores.configStore;
+
+var ConfigTableController = React.createClass({
+  getStoreState: function () {
+    return {
+      options: configStore.getOptions(),
+      loading: configStore.isLoading()
+    };
+  },
+
+  getInitialState: function () {
+    return this.getStoreState();
+  },
+
+  componentDidMount: function () {
+    configStore.on('change', this.onChange, this);
+  },
+
+  componentWillUnmount: function () {
+    configStore.off('change', this.onChange, this);
+  },
+
+  onChange: function () {
+    if (this.isMounted()) {
+      this.setState(this.getStoreState());
+    }
+  },
+
+  saveOption: function (option) {
+    Actions.saveOption(this.props.node, option);
+  },
+
+  deleteOption: function (option) {
+    Actions.deleteOption(this.props.node, option);
+  },
+
+  editOption: function (option) {
+    Actions.editOption(option);
+  },
+
+  cancelEdit: function () {
+    Actions.cancelEdit();
+  },
+
+  render: function () {
+    if (this.state.loading) {
+      return (
+        <div className="view">
+          <Components.LoadLines />
+        </div>
+      );
+    } else {
+      return (
+        <ConfigTable
+          onDeleteOption={this.deleteOption}
+          onSaveOption={this.saveOption}
+          onEditOption={this.editOption}
+          onCancelEdit={this.cancelEdit}
+          options={this.state.options}/>
+      );
+    }
+  }
+});
+
+var ConfigTable = React.createClass({
+  onSaveOption: function (option) {
+    this.props.onSaveOption(option);
+  },
+
+  onEditOption: function (option) {
+    this.props.onEditOption(option);
+  },
+
+  onDeleteOption: function (option) {
+    this.props.onDeleteOption(option);
+  },
+
+  onCancelEdit: function () {
+    this.props.onCancelEdit();
```

Comment:
Since you just  calling the function attached to the props, to simplify just pass the props down to the `ConfigOption` component then you don't need these functions.